src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1217,94 +1140,107 @@ protected void shutdownService() {
       this.runningThread = null;
       runningThread.interrupt();
     }
-    
-    /**
-     * Called when the queue has been updated and we may need to wake up the consumer thread.
-     */
-    protected void handleQueueUpdate() {
-      if (! startIfNotStarted()) {
-        Thread currRunningThread = runningThread;
-        if (currRunningThread != null) {
-          LockSupport.unpark(currRunningThread);
-        }
-      }
-    }
 
     @Override
     public void run() {
+      Worker worker = null;
       while (runningThread != null) {
         try {
-          TaskWrapper nextTask = getNextTask();
-          if (nextTask != null) {
-            switch (queuePriority) {
-              case High:
-                workerPool.runHighPriorityTask(nextTask);
-                break;
-              case Low:
-                workerPool.runLowPriorityTask(nextTask);
-                break;
-              default:
-                throw new UnsupportedOperationException();
+          worker = workerPool.getWorker();
+          if (worker != null) {
```

Comment:
I guess the reason I was grabbing the worker first now is because new tasks can be submitted (for example high priority) that would change the choosen task for execution while we wait on a worker.  This keeps the priority of execution accurate, while only having one thread to do the submissions of tasks to the worker threads.

That's the reason with multiple threads we could wait to grab the worker until we have a task ready to run.